src/travel-details-map-screen/use-get-live-service-journey-vehicles.ts

```diff
@@ -0,0 +1,68 @@
+import {useEffect, useState} from 'react';
+import {FetchResult, gql, useApolloClient} from '@apollo/client';
+import {VehiclePosition} from '@atb/api/types/generated/ServiceJourneyVehiclesQuery';
+import {useRealtimeMapEnabled} from '@atb/components/map/hooks/use-realtime-map-enabled';
+
+const DEFAULT_FETCH_POLICY = 'no-cache';
+
+export function useGetLiveServiceJourneyVehicles(
+  initialVehiclePosition?: VehiclePosition,
+) {
+  const [vehicles, setVehicles] = useState<VehiclePosition[]>([]);
+  const client = useApolloClient();
+  const realtimeMapEnabled = useRealtimeMapEnabled();
+  useEffect(() => {
+    if (initialVehiclePosition) {
+      setVehicles([initialVehiclePosition]);
+    }
+  }, [initialVehiclePosition]);
+
+  // Set up subscription to receive updates on vehicles
+  useEffect(() => {
+    if (!realtimeMapEnabled) return;
+
+    const subscription = client
+      .subscribe({
+        query: VEHICLE_UPDATES_SUBSCRIPTION,
+        fetchPolicy: DEFAULT_FETCH_POLICY,
+        variables: {
+          serviceJourneyId: initialVehiclePosition?.serviceJourney?.id,
+          includePointsOnLink: false,
+        },
+      })
+      .subscribe((fetchResult: FetchResult) => {
+        if (fetchResult?.data?.vehicleUpdates.length > 0) {
+          setVehicles(fetchResult?.data?.vehicleUpdates as VehiclePosition[]);
+        }
+      });
```

Comment:
Could some of this be moved to it's own function in the `api/vehicles.ts` file? Including the graphql query variables.